### DISCUSSION TITLE
[Q] How to set column width of table in streaming method

### DISCUSSION THREAD

**Discussion by lix-fs** [exceljs/exceljs#2378] [Q&A]
Hi, team
I have created a workbook by stream, such as `const excel = new StreamExcel({ filename: excelFilePath });`
But I can't set the column width By `const sheet = excel.addSheet(sheetName, { properties: { defaultColWidth: 30 } });`; `sheet.getColumn(col).width = 30;` dose't work neither. Even if I set the column width before commit each row, it doesn't work.
SteamExcel is a class encapsulated by exceljs;
```node.js
class StreamExcel {
  constructor(options) {
    const opt = Object.assign({
      filename: './streamed-workbook.xlsx',
      useStyles: true,
      useSharedStrings: true
    }, options);
    this.workbook = new excel.stream.xlsx.WorkbookWriter(opt);
    const now = new Date();
    this.workbook.created = now;
    this.workbook.modified = now;

    this.workbook.views = [
      {
        x: 0,
        y: 0,
        width: 10000,
        height: 20000,
        firstSheet: 0,
        activeTab: 1,
        visibility: 'visible'
      }
    ];
  }

addSheet(sheetName, options = null) {
    let sheet;
    if (options) {
      sheet = this.workbook.addWorksheet(sheetName, options);
    } else {
      sheet = this.workbook.addWorksheet(sheetName);
    }
    return sheet;
  }
}
```
thank you for help, plz

**Comment by skypesky** [comment]
@lix928  I can work with this.

```typescript
import { times } from 'lodash';

times(worksheet.columnCount, (index) => {
  const col: Partial<Column> = worksheet.getColumn(index + 1);
  col.width = 30;
});
```
**It is better to provide a complete test case!**

**Comment by lix-fs** [comment]
@skypesky I can't use ts `col: Partial<Column> `. I've try set the col.width in times, but don't work.
``` js
_.times(endColumn - startColumn, (index) => {
      const col = sheet.getColumn(index + startColumn);
      col.width = 30;
    });
sheet.commit();
await excel.commit();
```
this is a stream excel.

**Comment by ardabeyazoglu** [comment]
I have the same issue. It should work in case we define it before commiting.

**Comment by hlerebours** [comment]
Same issue here. Looks like the only way is to set the widths before adding rows, using:
```
sheet.columns = _.times(columnCount, () => ({ width: 30 }));
```
In my case, it's really not handy because I want to fit the columns to the content, which is not known in advance (streamed too). :disappointed:

**Comment by natefabian18** [comment]
Was a solution ever found for this to allow columns to be set dynamically as the data is streamed in? Also having the same issue.

**Comment by hlerebours** [comment]
I ended up consuming the first 100 rows of my stream, hoping they would be somewhat representative, to compute the columns widths, then inserting these rows and the rest of the stream to the worksheet... Hardly satisfying.
